### ISSUE TITLE
Plugins (eg. "watch-run") for MultiCompiler instance not works

### ISSUE THREAD

**Issue by shyr1punk** [webpack/webpack#5570]
**Do you want to request a *feature* or report a *bug*?**
I don't know, its a bug or feature. Or needs docs improvement

**What is the current behavior?**

I create a MultiCompiler instance and add "watch-run" plugin:
```
const multiCompiler = webpack([{...}, {...}]);

multiCompiler.plugin('watch-run', function(watching, cb) {
  console.log('watch-run'); // don't print
  cb();
});

const watcher = multiCompiler.watch({}, (err, stats) => {
  console.log('Rebuild complete ' + new Date().toLocaleString()); //print
});
```
And this plugin's don't executes after project file change.

If I add plugin for compiler instance in `MultiCompiler.compilers` it's works:

```
const multiCompiler = webpack([{...}, {...}]);

multiCompiler.compilers[0].plugin('watch-run', function(watching, cb) {
  console.log('watch-run'); // print
  cb();
});

...
```

**If the current behavior is a bug, please provide the steps to reproduce.**

https://github.com/shyr1punk/webpack-issue

**What is the expected behavior?**

Plugins in MultiCompiler instance executes 

**If this is a feature request, what is motivation or use case for changing the behavior?**

I suggest:
1) Add docs for multi compiler plugin usage OR
2) Make it possible add plugins for MultiCompiler instance with binding to each complilers OR
3) I do something wrong?

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

Node.js 8.1.4
Webpack 2.7.0
MacOS X 10.11.6

**Comment by sokra** [comment]
I guess this is not implemented for the MultiCompiler.

Feel free to send a PR.

**Comment by asborisov** [comment]
Based on https://github.com/webpack/webpack/blob/master/lib/MultiCompiler.js only `done` and `invalid` hook implemented. You can create PR or just assign new plugin for each of the compilers:
```js
multiCompiler.compilers.forEach(compiler => {
   compiler.plugin('watch-run', ...);
});
```

**Comment by okcoker** [comment]
@sokra is there a reason only "invalid" and "done" are implemented for `MultiCompiler`? I've submitted a PR for webpack-hot-middleware (https://github.com/glenjamin/webpack-hot-middleware/pull/260) to use the "invalid" event over "compile" since `Compiler` and `MultiCompiler` both use "invalid" but I'm curious if there's a reason behind only having the 2 events.

**Comment by ooflorent** [comment]
Fixed on #6209 and released in webpack 4.0.0-alpha.3.
